packages/flutter/lib/src/widgets/linked_text.dart

```diff
@@ -0,0 +1,614 @@
+        _cleanTextLinkerMatches(
+          _TextLinkerMatch.fromTextLinkers(textLinkers, spansText),
+        );
+
+    final (Iterable<InlineSpan> output, Iterable<_TextLinkerMatch> _, Iterable<TapGestureRecognizer> recognizers) =
+        _linkSpansRecurse(spansWithLength, textLinkerMatches, 0);
+    return (output, recognizers);
+  }
+
+  static List<_TextLinkerMatch> _cleanTextLinkerMatches(Iterable<_TextLinkerMatch> textLinkerMatches) {
+    final List<_TextLinkerMatch> nextTextLinkerMatches = textLinkerMatches.toList();
+
+    // Sort by start.
+    nextTextLinkerMatches.sort((_TextLinkerMatch a, _TextLinkerMatch b) {
+      return a.textRange.start.compareTo(b.textRange.start);
+    });
+
+    int lastEnd = 0;
+    nextTextLinkerMatches.removeWhere((_TextLinkerMatch textLinkerMatch) {
+      // Return empty ranges.
+      if (textLinkerMatch.textRange.start == textLinkerMatch.textRange.end) {
+        return true;
+      }
+
+      // Remove overlapping ranges.
+      final bool overlaps = textLinkerMatch.textRange.start < lastEnd;
+      if (!overlaps) {
+        lastEnd = textLinkerMatch.textRange.end;
+      }
+      return overlaps;
+    });
+
+    return nextTextLinkerMatches;
+  }
+
+  static (Iterable<InlineSpan>, Iterable<_TextLinkerMatch>, Iterable<TapGestureRecognizer>) _linkSpansRecurse(Iterable<({InlineSpan span, int length})> spansWithLength, Iterable<_TextLinkerMatch> textLinkerMatches, int index) {
+    final List<InlineSpan> output = <InlineSpan>[];
+    Iterable<_TextLinkerMatch> nextTextLinkerMatches = textLinkerMatches;
+    final List<TapGestureRecognizer> recognizers = <TapGestureRecognizer>[];
+    int nextIndex = index;
+    for (final (span: InlineSpan span, length: int length) in spansWithLength) {
+      final (InlineSpan childSpan, Iterable<_TextLinkerMatch> childTextLinkerMatches, Iterable<TapGestureRecognizer> childRecognizers) = _linkSpanRecurse(
+        span,
+        nextTextLinkerMatches,
+        nextIndex,
+      );
+      output.add(childSpan);
+      nextTextLinkerMatches = childTextLinkerMatches;
+      recognizers.addAll(childRecognizers);
+      nextIndex += length;
+    }
+
+    return (output, nextTextLinkerMatches, recognizers);
+  }
+
+  // index is the index of the start of `span` in the overall flattened tree
+  // string.
+  //
+  // The TapGestureRecognizers must be disposed by an owning widget.
+  static (InlineSpan, Iterable<_TextLinkerMatch>, Iterable<TapGestureRecognizer>) _linkSpanRecurse(InlineSpan span, Iterable<_TextLinkerMatch> textLinkerMatches, int index) {
+    if (span is! TextSpan) {
+      return (span, textLinkerMatches, <TapGestureRecognizer>[]);
+    }
+
+    final List<InlineSpan> nextChildren = <InlineSpan>[];
+    final List<TapGestureRecognizer> recognizers = <TapGestureRecognizer>[];
+    List<_TextLinkerMatch> nextTextLinkerMatches = <_TextLinkerMatch>[...textLinkerMatches];
+    int lastLinkEnd = index;
+    if (span.text?.isNotEmpty ?? false) {
```

Comment:
It looks like the linker can't across textspan boundary? Can you also document this behavior in doc string?

Reply:
but I think it should be doable if we really want to implement it?

Reply:
It does work across TextSpans, if you mean like this:

```dart
                LinkedText(
                  onTap: _onTapUrl,
                  spans: [
                    TextSpan(
                      text: 'ex',
                    ),
                    TextSpan(
                      text: 'ample.',
                    ),
                    TextSpan(
                      text: 'com',
                    ),
                  ],
                ),
```
<img width="115" alt="Screenshot 2023-08-30 at 9 28 01 AM" src="https://github.com/flutter/flutter/assets/389558/909dda93-b43b-49d3-85fd-7a3453e2b61f">

It produces three separate InlineLinkSpans ("ex", "emple.", "com") that all link to "example.com".